src/testing/TGV_scaling.cpp

```diff
@@ -0,0 +1,381 @@
+{
+    //get local CFL
+    const unsigned int n_dofs_cell = nstate*pow(poly_degree+1,dim);
+    const unsigned int n_quad_pts = pow(poly_degree+1,dim);
+    std::vector<dealii::types::global_dof_index> dofs_indices1 (n_dofs_cell);
+
+    double cfl_min = 1e100;
+    std::shared_ptr < Physics::PhysicsBase<dim, nstate, double > > pde_physics_double  = PHiLiP::Physics::PhysicsFactory<dim,nstate,double>::create_Physics(dg->all_parameters);
+    for (auto cell = dg->dof_handler.begin_active(); cell!=dg->dof_handler.end(); ++cell) {
+        if (!cell->is_locally_owned()) continue;
+        cell->get_dof_indices (dofs_indices1);
+        std::vector< std::array<double,nstate>> soln_at_q(n_quad_pts);
+        for (unsigned int iquad=0; iquad<n_quad_pts; ++iquad) {
+            for (int istate=0; istate<nstate; istate++) {
+                soln_at_q[iquad][istate]      = 0;
+            }
+        }
+        for(unsigned int iquad=0; iquad<n_quad_pts; iquad++){
+          dealii::Point<dim> qpoint = dg->volume_quadrature_collection[poly_degree].point(iquad);
+            for(unsigned int idof=0; idof<n_dofs_cell; idof++){
+                const unsigned int istate = dg->fe_collection[poly_degree].system_to_component_index(idof).first;
+                soln_at_q[iquad][istate] += dg->solution[dofs_indices1[idof]] * dg->fe_collection[poly_degree].shape_value_component(idof, qpoint, istate);
+            }
+        }
+
+        std::vector< double > convective_eigenvalues(n_quad_pts);
+        for (unsigned int isol = 0; isol < n_quad_pts; ++isol) {
+            convective_eigenvalues[isol] = pde_physics_double->max_convective_eigenvalue (soln_at_q[isol]);
+        }
+        const double max_eig = *(std::max_element(convective_eigenvalues.begin(), convective_eigenvalues.end()));
+
+        const double max_eig_mpi = dealii::Utilities::MPI::max(max_eig, mpi_communicator);
+        double cfl = 0.1 * delta_x/max_eig_mpi;
+        if(cfl < cfl_min)
+            cfl_min = cfl;
+
+    }
+    return cfl_min;
+}
+
+template <int dim, int nstate>
+int EulerTaylorGreenScaling<dim, nstate>::run_test() const
+{
+    // using Triangulation = dealii::parallel::distributed::Triangulation<dim>;
+    // std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation> (mpi_communicator);
+    using Triangulation = dealii::parallel::distributed::Triangulation<dim>;
+    std::shared_ptr<Triangulation> grid = std::make_shared<Triangulation>(
+        MPI_COMM_WORLD,
+        typename dealii::Triangulation<dim>::MeshSmoothing(
+            dealii::Triangulation<dim>::smoothing_on_refinement |
+            dealii::Triangulation<dim>::smoothing_on_coarsening));
+
+    using real = double;
+
+    PHiLiP::Parameters::AllParameters all_parameters_new = *all_parameters;  
+    double left = 0.0;
+    double right = 2 * dealii::numbers::PI;
+
+    const unsigned int n_refinements = all_parameters->flow_solver_param.number_of_grid_elements_per_dimension;
```

Comment:
maybe flow_solver_param.number_of_mesh_refinements would be better?

Reply:
fixed